src/Resources/config/core.php

```diff
@@ -0,0 +1,235 @@
+            ])
+
+        ->alias(BreadcrumbsBuilder::class, 'sonata.admin.breadcrumbs_builder')
+
+        ->alias(BreadcrumbsBuilderInterface::class, 'sonata.admin.breadcrumbs_builder')
+
+        // Services used to format the label, default is sonata.admin.label.strategy.noop
+
+        ->set('sonata.admin.label.strategy.bc', BCLabelTranslatorStrategy::class)
+            ->public()
+
+        ->alias(BCLabelTranslatorStrategy::class, 'sonata.admin.label.strategy.bc')
+
+        ->set('sonata.admin.label.strategy.native', NativeLabelTranslatorStrategy::class)
+            ->public()
+
+        ->alias(NativeLabelTranslatorStrategy::class, 'sonata.admin.label.strategy.native')
+
+        ->alias(LabelTranslatorStrategyInterface::class, 'sonata.admin.label.strategy.native')
+
+        ->set('sonata.admin.label.strategy.noop', NoopLabelTranslatorStrategy::class)
+            ->public()
+
+        ->alias(NoopLabelTranslatorStrategy::class, 'sonata.admin.label.strategy.noop')
+
+        ->set('sonata.admin.label.strategy.underscore', UnderscoreLabelTranslatorStrategy::class)
+            ->public()
+
+        ->alias(UnderscoreLabelTranslatorStrategy::class, 'sonata.admin.label.strategy.underscore')
+
+        ->set('sonata.admin.label.strategy.form_component', FormLabelTranslatorStrategy::class)
+            ->public()
+
+        ->alias(FormLabelTranslatorStrategy::class, 'sonata.admin.label.strategy.form_component')
+
+        // NEXT_MAJOR: Remove this service.
+        ->set('sonata.admin.translator.extractor.jms_translator_bundle', DeprecatedAdminExtractor::class)
+            ->public()
+            ->tag('jms_translation.extractor', [
+                'alias' => 'sonata_admin',
+            ])
+            ->deprecate(sprintf(
+                'The service "%%service_id%%" is deprecated in favor of the "%s" service since version 3.72 and will be removed in 4.0.',
+                Sonata\AdminBundle\Translator\Extractor\AdminExtractor::class
+            ))
+            ->args([
+                new ReferenceConfigurator('sonata.admin.pool'),
+                (new ReferenceConfigurator('logger'))->nullOnInvalid(),
+            ])
+            ->call('setBreadcrumbsBuilder', [
+                new ReferenceConfigurator('sonata.admin.breadcrumbs_builder'),
+            ])
+
+        ->set(AdminExtractor::class)
+            ->tag('translation.extractor', [
+                'alias' => 'sonata_admin',
+            ])
+            ->args([
+                new ReferenceConfigurator('sonata.admin.pool'),
+                new ReferenceConfigurator('sonata.admin.breadcrumbs_builder'),
+            ])
+
+        // NEXT_MAJOR: Remove this service.
+        ->set('sonata.admin.controller.admin', HelperController::class)
+            ->public()
+            ->deprecate('The controller service "%service_id%" is deprecated in favor of several action services since version 3.38.0 and will be removed in 4.0.')
```

Comment:
```suggestion
            ->deprecate('The controller service "%service_id%" is deprecated in favor of several action services since sonata-project/admin-bundle 3.38.0 and will be removed in 4.0.')
```